### PR TITLE
add telemetry for LLM selection and CTA clicks

### PR DESCRIPTION
**Pull request by abeatrix**
feat: add telemetry for LLM selection and CTA clicks

Track events when users click on the Cody Pro upgrade CTA and when they select an AI model from the dropdown.

## Test plan

https://github.com/sourcegraph/cody/assets/68532117/976a5981-db66-4697-b115-e46859ba7643

### CODE REVIEW

File: vscode/webviews/Components/ChatModelDropdownMenu.tsx [hunk 1]
```diff
@@ -25,11 +25,20 @@ export const ChatModelDropdownMenu: React.FunctionComponent<ChatModelDropdownMen
     const handleChange = useCallback(
         (event: any): void => {
             if (showCodyProBadge) {
-                console.log('Cody Pro badge clicked')
                 getVSCodeAPI().postMessage({ command: 'links', value: 'https://sourcegraph.com/cody/subscription' })
+                getVSCodeAPI().postMessage({
+                    command: 'event',
+                    eventName: 'CodyVSCodeExtension:upgradeCTA:clicked',
```
Comment (chenkc805) [inline vscode/webviews/Components/ChatModelDropdownMenu.tsx:31]: @abeatrix I'm seeing this event labeled as `CodyVSCodeExtension:upsellUsageLimitCTA:clicked` in the video, which is different from what I see here. 

Let's change this event to say `CodyVSCodeExtension:upgradeLLMChoiceCTA:clicked`

**Review by chenkc805 — CHANGES_REQUESTED** [review]
Just left a small change request. Other LGTM

File: vscode/webviews/Components/ChatModelDropdownMenu.tsx (hunk 1 above)
Comment (abeatrix) [reply vscode/webviews/Components/ChatModelDropdownMenu.tsx]: renamed

**Review by chenkc805 — APPROVED** [review]
LGTM

**Review by philipp-spiess — APPROVED** [review]

**Review by philipp-spiess — APPROVED** [review]

**Pull request merged**